sig-cluster-lifecycle/charter.md

```diff
@@ -0,0 +1,65 @@
+# SIG Cluster Lifecycle Charter
+
+## Scope
+
+SIG Cluster Lifecycle’s objective is to improve the deployment, configuration management, and upgrade story for Kubernetes clusters and relevant infrastructure.
+
+### In scope
+
+The following topics fall under ownership of this SIG:
+
+- Improving the Kubernetes UX for cluster administration
+- Kubernetes cluster lifecycle related features and issues
+- Tools that assist in bootstrapping, mutating, and upgrading Kubernetes control plane components. 
```

Comment:
We discussed addons in the sig meeting and folks weren't in agreement on adding it here. 

I think @justinsb thought this had enough wiggle room to cover them, but it would be better if we had consensus of whether they were in or out and made that explicit.

Reply:
At least the following the tools owned by SIG cluster lifecycle have created "add-on" mechanisms for configuring self-hosted components (e.g., kube-proxy, kube-dns) and other API resources:
* kube-up: https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/addon-manager
* kops: https://github.com/kubernetes/kops/blob/master/docs/addon_manager.md
* minikube: https://github.com/kubernetes/minikube/blob/master/pkg/minikube/assets/addons.go
* kubeadm: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/phases/addons.go

I also kicked off this thread last year:
https://groups.google.com/d/msg/kubernetes-dev/wtXaoHOiSfg/QFW5Ca9YBgAJ